sqlglot/parser.py

```diff
@@ -3426,7 +3426,11 @@ def _parse_types(
         return this
 
     def _parse_struct_types(self) -> t.Optional[exp.Expression]:
-        this = self._parse_type() or self._parse_id_var()
+        if self._next and self._next.token_type == TokenType.COLON:
```

Comment:
not sure i understand this change, struct<interval double> is also legal, the semi colon is optional

Reply:
The issue here was that `interval: int` was parsed as an `Interval` with a `Placeholder` child. So the idea is: if we have <something> <colon> ... then this is probably a "column definition" (in parser terms), and so we want to parse <something> as an id var to avoid this scenario. `STRUCT(interval)` works fine if that's what you mean.

Reply:
what about struct(interval double)

Reply:
```python
>>> sqlglot.parse_one("struct(interval double)")
(STRUCT expressions:
  (INTERVAL this:
    (COLUMN this:
      (IDENTIFIER this: double, quoted: False))))
```

I see what you mean, didn't remember that `STRUCT(<name> <type>)` was valid

Reply:
What if we dodge `_parse_interval` in `_parse_type` with a flag then?

```python
def _parse_type(self, parse_interval: bool = True) -> ...
```

Reply:
sure

Reply:
Done